torch/linalg/__init__.py

```diff
@@ -1419,69 +1481,71 @@
-    >>> torch.linalg.pinv(a)
+    >>> A = torch.randn(3, 3)
+    >>> torch.linalg.pinv(A)
     tensor([[ 0.2971, -0.4280, -2.0111],
             [-0.0090,  0.6426, -0.1116],
             [-0.7832, -0.2465,  1.0994]])
-    >>> torch.linalg.pinv(a, rcond)
+    >>> torch.linalg.pinv(A, rcond)
     tensor([[-0.2672, -0.2351, -0.0539],
             [-0.0211,  0.6467, -0.0698],
             [-0.4400, -0.3638, -0.0910]])
 
     Matrix-wise rcond example
-    >>> a = torch.randn(5, 6, 2, 3, 3)
+    >>> A = torch.randn(5, 6, 2, 3, 3)
     >>> rcond = torch.rand(2)  # different rcond values for each matrix in a[:, :, 0] and a[:, :, 1]
-    >>> torch.linalg.pinv(a, rcond)
+    >>> torch.linalg.pinv(A, rcond)
     >>> rcond = torch.randn(5, 6, 2) # different rcond value for each matrix in 'a'
-    >>> torch.linalg.pinv(a, rcond)
+    >>> torch.linalg.pinv(A, rcond)
+
+.. _defined algebraically:
+    https://en.wikipedia.org/wiki/Moore%E2%80%93Penrose_inverse#Existence_and_uniqueness
+.. _through the SVD:
+    https://en.wikipedia.org/wiki/Moore%E2%80%93Penrose_inverse#Singular_value_decomposition_(SVD)
 """)
 
 solve = _add_docstr(_linalg.linalg_solve, r"""
-linalg.solve(input, other, *, out=None) -> Tensor
-
-Computes the solution of a square system  of linear equations with unique solution. Supports batched inputs.
-
-This method accepts both matrices and vectors as the right-hand side of the linear equation.
-We will denote by `*` zero or more batch dimensions.
+linalg.solve(A, B, *, out=None) -> Tensor
 
-- If :attr:`input`, :attr:`other` have shapes `(*, n, n)`, `(*, n)`, for every matrix :math:`A`
-  in :attr:`input` and vector :math:`b` in :attr:`other`, this function returns a vector :math:`x` such that
+Computes the solution of a square system of linear equations with unique solution.
 
-  .. math:: Ax = b
+Letting :math:`\mathbb{K}` be :math:`\mathbb{R}` or :math:`\mathbb{C}`,
+this function computes a solution :math:`X \in \mathbb{K}^{n \times k}` of the linear system associated to
+:math:`A \in \mathbb{K}^{n \times n}, B \in \mathbb{K}^{m \times k}`, which is defined as
 
-  The returned tensor have shape `(*, n)`.
-- If :attr:`input`, :attr:`other` have shapes `(*, n, n)`, `(*, n, k)`, for every pair of matrices
-  :math:`A` in :attr:`input` and :math:`B` in :attr:`other`, this function returns a matrix :math:`X` such that
+.. math:: AX = B
 
-  .. math:: AX = B
+Supports inputs of float, double, cfloat and cdouble dtypes.
+Also supports batched inputs, and, if the input is batched, the output is batched with the same dimensions.
 
-  The returned tensor have shape `(*, n, k)`.
-- If none of the above apply and :attr:`input`, :attr:`other` have shapes `(*, n, n)`, `(n,)` (resp. `(n, k)`),
-  the vector (resp. matrix) :attr:`other` is broadcasted to have shape `(*, n)` (resp. `(*, n, k)`).
-  This function then  returns the solution of the resulting batch of systems of linear equations.
+Denote by `*` zero or more batch dimensions.
```

Comment:
I would move this to a "returns" section (like some other entries)

Instead of "Denote by..." use a let expression like "Letting \`*` be zero or more batch dimensions, the shape of the output is:"

Reply:
I left this here because it's this odd behaviour of being able to pass a vector or a matrix, and then the even weirder broadcasting behaviour. I think we should just accept matrices on both sides with the same batch dimensions, but numpy compat...

Denote changed.

About the organisation, I could split it as I did for SVD and QR (full vs reduced version), but I think it is not worth it. I think we should leave this here to document the odd casting behaviour. I would just move things to the `Returns` section when they are somewhat unimportant / they follow from the mathematical treatment. I would not want to move this, as it's just "an oddity" that we support, and it could bite people that are not aware of it.